app/javascript/shared/components/plateScanValidators.js

```diff
@@ -375,6 +375,28 @@ const checkPlateWithSameReadyLibrarySubmissions = (cached_submission_ids) => {
   }
 }
 
+// Returns a validator that ensures that the scanned plate does not have an unacceptable plate purpose.
+// plate: The current plate being scanned
+const checkForUnacceptablePlatePurpose = (acceptable_purposes) => {
+  return (plate) => {
+    if (!acceptable_purposes || acceptable_purposes.length == 0) {
+      // return valid if no acceptable purposes are provided
+      return validScanMessage()
+    } else if (acceptable_purposes.includes(plate.purpose.name)) {
```

Comment:
Wondering whether we should guard against the plate purpose name not being available...

Reply:
added, thx